src/amp.js

```diff
@@ -67,6 +69,8 @@ chunk(self.document, function initial() {
       // Core services.
       installRuntimeServices(self);
       fontStylesheetTimeout(self);
+      installViewerServiceForDoc(ampdoc);
```

Comment:
Hmm. I'm a bit concerned about leaking lots of services outside the adopt routine. Let's whiteboard and see if we can find a better way to override. If anything, testing will be more challenging this way. Perhaps we can explore the path we did with shadow runtime and `adoptEnv` pattern?